src/librustdoc/html/render/cache.rs

```diff
@@ -237,14 +237,14 @@ fn get_index_type_name(clean_type: &clean::Type, accept_generic: bool) -> Option
         }
         clean::DynTrait(ref bounds, _) => get_index_type_name(&bounds[0].trait_, accept_generic),
         clean::Generic(s) if accept_generic => Some(s),
+        clean::Primitive(clean::PrimitiveType::Never) => None,
```

Comment:
I don't think this is right. `clean::Primitive(clean::PrimitiveType::Never)` _was_ used before this change; I think it was used for things like the `primitive.never.html` page. This change seems like it'll break searching for `never`. I think you can just remove this line and let the existing `clean::Primitive(ref p)` match arm handle it.

Reply:
I can confirm that it works perfectly fine as is.

Reply:
It was unneeded as well, removed this case too.